src/ts/components/listbox.ts

```diff
@@ -0,0 +1,173 @@
+import {UIInstanceManager} from '../uimanager';
+import {Button, ButtonConfig} from './button';
+import {ListSelector, ListSelectorConfig} from './listselector';
+import {DOM} from '../dom';
+import {Timeout} from '../timeout';
+
+export interface ListBoxConfig extends ListSelectorConfig {
```

Comment:
The `ListBox` should be a component analogous to `SelectBox`, i.e. it would be used for example as a child inside a `SettingsPanel`. A `SettingsPanel` can contain one or many `ListBox` instances. A `SelectBox` also does not have a cointainer or cares about hiding, that's the business of whatever `Container` the `SelectBox` or `ListBox` night be put into. This is also important in regards to paging (#119), where a `SettingsPanel` will be able to contain multiple `SettingsPanel`s that can be navigated.

It seems what you have created here is a hybrid of a `SettingsPanel` and a `ListBox`. This goes against the design principle of component reuse and composition.